### ISSUE TITLE
The process for setting up SSL on the local server on Windows

### ISSUE THREAD

**Issue by SimonBiggs** [libreapp/libreapp#4]
https://www.starwindsoftware.com/blog/using-the-microsoft-certificate-authority-to-get-rid-of-those-self-signed-certs

**Comment by darcymason** [comment]
Looks interesting.  So certificate is generated by local I.T. and stored in config of the local minimal server?

**Comment by SimonBiggs** [comment]
That's the plan. Do you think that would work? It seems like quite a big ask, do you think it is an insurmountable request from users who want the local server?

**Comment by SimonBiggs** [comment]
... I guess the question is, is it a deal breaker?

**Comment by darcymason** [comment]
I don't think it is necessarily a deal-breaker - but it would require very clear directions (or hand-holding).  Some may not be able to do it (I.T. doesn't want to allow it), in which case an alternative may have to be offered (e.g. upload buttons for files).

**Comment by SimonBiggs** [comment]
There certainly needs to be an alternative. The local-server option needs to be the advanced case, but everything needs to be built with the concept of "no server available" first.

**Comment by SimonBiggs** [comment]
This is very interesting:

https://letsencrypt.org/docs/certificates-for-localhost/

https is allowed to communicate with the address 127.0.0.1! But only that address. I think that's the solution!

**Comment by SimonBiggs** [comment]
Ahh seems like websockets are allowed:

https://news.ycombinator.com/item?id=17361007

**Comment by SimonBiggs** [comment]
Or maybe I have been thinking about this all wrong. See the following:

https://bugs.chromium.org/p/chromium/issues/detail?id=418482#c8

And therefore the following:

* https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging
* https://developer.chrome.com/extensions/nativeMessaging

**Comment by SimonBiggs** [comment]
Hmm, might be a bit painful to use:

https://stackoverflow.com/a/27260676/3912576

... but worth the pain I think ...
